File: src/main/python/binaryTreeRightSideView/BinaryTreeRightSideView.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> list[int]:
        result = []
        if root is None:
            return result
        tmp = [[root]]
        while root is not None:
            cur_list = tmp.pop(0)
            result.append(cur_list[-1].val)
            next_tmp = []
            for i in range(len(cur_list)):
                cur = cur_list[i]
                if cur.left is not None:
                    next_tmp.append(cur.left)
                if cur.right is not None:
                    next_tmp.append(cur.right)
            if len(next_tmp) == 0:
                break
            tmp.append(next_tmp)
        return result

File: src/main/kotlin/me/bossm0n5t3r/leetcode/binaryTreeRightSideView/BinaryTreeRightSideView.kt
package me.bossm0n5t3r.leetcode.binaryTreeRightSideView

import me.bossm0n5t3r.leetcode.utils.TreeNode
import java.util.LinkedList

class BinaryTreeRightSideView {
    class Solution {
        fun rightSideView(root: TreeNode?): List<Int> {
            if (root == null) return emptyList()
            val queue =
                LinkedList<TreeNode>().apply {
                    this.add(root)
                }
            val result = mutableListOf<Int>()
            while (queue.isNotEmpty()) {
                val size = queue.size
                result.add(queue.last().`val`)
                repeat(size) {
                    val cur = queue.poll()
                    val curLeft = cur.left
                    val curRight = cur.right
                    if (curLeft != null) {
                        queue.add(curLeft)
                    }
                    if (curRight != null) {
                        queue.add(curRight)
                    }
                }
            }
            return result
        }
    }
}
